src/js/_enqueues/lib/nav-menu.js

```diff
@@ -876,6 +876,35 @@
 					$( '#submit-customlinkdiv' ).trigger( 'click' );
 				}
 			});
+
+			$('#submit-customlinkdiv').on('click', function(e) {
+				var urlInput = $('#custom-menu-item-url'),
+					url = urlInput.val(),
+					errorMessage = $('#custom-url-error'),
+					urlWrap = $('#menu-item-url-wrap');
+			
+				// Hide the error message initially
+				errorMessage.hide();
+				urlWrap.removeClass('has-error');
+			
+				if ('' === url || 'https://' === url || 'http://' === url) {
+					e.preventDefault();
+					urlInput.addClass('form-invalid')
+						.attr('aria-invalid', 'true')
+						.attr('aria-describedby', 'custom-url-error');
+			
+					// Show the error message and add the class for extra margin
+					errorMessage.show();
+					urlWrap.addClass('has-error');
+			
+					// Announce error message via screen reader
+					wp.a11y.speak('Invalid URL. Please enter a valid URL.', 'assertive');
```

Comment:
This needs to be a translatable string. Also, the text doesn't match the actual validation rule - which is really only checking whether the value is logically empty. (Either blank, or a protocol with no values.)

The message should probably be something more like "No content entered. Please enter valid link." I don't want to say URL, because you can enter values that aren't URLs - e.g., `#content` could be used to define a skip link, or `#open-modal` could be used to open a modal on the page. Because these fields have historically allowed almost any content, we really can't expect a URL now, and the message should reflect that.